### ISSUE TITLE
Removing non-existent property results in error.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2033]
|<img alt="rfmohr@...'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| rfmohr@... reported|
|-|-|
|Trac ID|trac#2033|
|Type|defect|
|Created|2006-08-29 16:57:48Z|

I ran the litmus-0.10.5 property tests after applying the xattr prefix fix mentioned in [#1828](https://github.com/twisted/twisted/issues/1828).  There were several failures:
```
-> running `props':
 0. init.................. pass
 1. begin................. pass
...
12. propget............... pass
13. propreplace........... FAIL (PROPPATCH on `/home/bar/litmus/prop2': 500 Internal Server Error)
14. propget............... FAIL (Property {http://webdav.org/neon/litmus/}prop9 had value value9, expected newvalue9)
15. propnullns............ pass
16. propget............... FAIL (No value given for property {}nonamespace)
17. prophighunicode....... pass
...
25. finish................ pass
<- summary for `props': of 26 tests run: 23 passed, 3 failed. 88.5%
```
I saw these messages in the server log (full output is in attachment):
```
[Tue Aug 29 12:38:21 2006] [error] [client 127.0.0.1] "Can't find property (u'http://webdav.org/neon/litmus/', u'prop4') for resource /home/bar/litmus/prop2"
[Tue Aug 29 12:38:22 2006] [error] [client 127.0.0.1] Traceback (most recent call last):
<snip>
[Tue Aug 29 12:38:22 2006] [error] [client 127.0.0.1]   File "/usr/lib/python2.3/site-packages/twisted/python/failure.py", line 259, in raiseException
[Tue Aug 29 12:38:22 2006] [error] [client 127.0.0.1]     raise self.type, self.value, self.tb
[Tue Aug 29 12:38:22 2006] [error] [client 127.0.0.1] exceptions.KeyError: u'user.{http:%2F%2Fwebdav.org%2Fneon%2Flitmus%2F}prop0'
```
It seems that the error occurs when an attempt is made to remove a property that does not exist.  According to RFC 2518 Section 12.13.1, "Specifying the removal of a property that does not exist is not an error."

The KeyError exception needs to be caught and handled.  I will post a possible patch once I have one.

Attachments:

* [litmus-output](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ffb/ffba58386f77f9616685b1558613aba9a9cfacce/d98376958e331345a1f01eb6a6a2e1428be0984a) (1183 bytes) - added by rfmohr@... on 2006-08-29 16:58:41Z - litmus-0.10.5 props test output
* [error-log](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ffb/ffba58386f77f9616685b1558613aba9a9cfacce/d026cbb4395cc7935c455c0430c7577dd4096a15) (4979 bytes) - added by rfmohr@... on 2006-08-29 16:59:49Z - Apache error_log output
* [remove-props.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ffb/ffba58386f77f9616685b1558613aba9a9cfacce/adf73b30ef710f29c135b24b6c14ff6ccee9e092.patch) (529 bytes) - added by rfmohr@... on 2006-08-29 17:23:31Z - Patch for xattrPropertyStore when removing non-existent property.

<details><summary>Searchable metadata</summary>

```
trac-id__2033 2033
type__defect defect
reporter__rfmohr____ rfmohr@...
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2_dav web2.dav
keywords__dav dav
time__1156870668000000 1156870668000000
changetime__1158093381000000 1158093381000000
version__None None
owner__ 
cc__wsanchez
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="rfmohr@...'s avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|rfmohr@... commented|
|-|-|

I have attached a patch for the xattrPropertyStore class which will catch any KeyError exceptions when removing a property that does not exist.

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_2"></a>|@jyknight set owner to @wsanchez|
|-|-|

Assigning to wsanchez

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez commented|
|-|-|

Branch: dav-remove-props-2033

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez set status to `closed`|
|-|-|

(In [18100]) Don't raise when attempting to delete a property that doesn't exist.
Submitted by rfmohr@...
Reviewed by wsanchez
Fixes [#2033](https://github.com/twisted/twisted/issues/2033)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
